src/Shell/Task/ModelTask.php

```diff
@@ -844,8 +844,8 @@ public function bakeTable($model, array $data = [])
             $namespace = $this->_pluginNamespace($this->plugin);
         }
 
-        $name = $model->alias();
-        $entity = $this->_entityName($model->alias());
+        $name = (!empty($this->params['model-prefix'])?$this->_camelize($this->params['model-prefix']):'') . $model->alias();
+        $entity = $this->_entityName((!empty($this->params['model-prefix'])?$this->_camelize($this->params['model-prefix']):'') . $model->alias());
```

Comment:
He could just reuse $name from above which does exactly the same, right?
And just say

    $entity = $this->_entityName($name);